SQL_files/baseball_batting_average.sql

```diff
@@ -0,0 +1,52 @@
+HOW ERRORS;
+SHOW WARNINGS;
+COMMIT;
+
+/* 
+use baseball schema 
+Note: BA = avg(hit/atBat)
+*/
+USE baseball;
+
+/*
+Table: historical table (BA overall) 
+Note: COUNT(batter) = number of times batter played and number of games (batter shows up once per game)
+*/
+DROP TABLE IF EXISTS BATTERS_HISTORICAL;
+CREATE TABLE BATTERS_HISTORICAL AS
+(SELECT batter AS Batter, COUNT(batter) AS Batters_Count, SUM(Hit) AS Hit_Sum, SUM(atBat) AS atBat_Sum, SUM(Hit)/NULLIF(SUM(atBat), 0) AS Batting_AVG
+FROM baseball.batter_counts GROUP BY Batter);
+
+SELECT *
+FROM baseball.BATTERS_HISTORICAL;
+
+/*
+Table: annual table (BA annually) 
+*/
+DROP TABLE IF EXISTS BATTERS_ANNUALLY;
+CREATE TABLE BATTERS_ANNUALLY AS
+(SELECT baseball.batter_counts.batter AS Batter, REPLACE(YEAR(baseball.game.local_date),',','') AS The_Year, 
```

Comment:
Instead of replacing the local date with year, it may be more concise to SELECT ...YEAR(local_date) as The_Year

Reply:
I did that just for stylistic purposes so I could remove the comma in the year haha